contrib/gimp.lua

```diff
@@ -0,0 +1,247 @@
+
+    This script provides another storage (export target) for darktable.  Selected
+    images are exported in the specified format to temporary storage.  GIMP is launched
+    and opens the files.  After editing, the exported images are overwritten to save the 
+    changes.  When GIMP exits, the exported files are moved into the current collection
+    and imported into the database.  The imported files then show up grouped with the 
+    originally selected images.
+
+    ADDITIONAL SOFTWARE NEEDED FOR THIS SCRIPT
+    * GIMP - http://www.gimp.org
+
+    USAGE
+    * require this script from your main lua file
+    * select an image or images for editing with GIMP
+    * in the export dialog select "Edit with GIMP" and select the format and bit depth for the
+      exported image
+    * Press "export"
+    * Edit the image with GIMP then save the changes with File->Overwrite....
+    * Exit GIMP
+    * The edited image will be imported and grouped with the original image
+
+    CAVEATS
+    * Developed and tested on Ubuntu 14.04 LTS with darktable 2.0.3 and GIMP 2.9.3 (development version with
+      > 8 bit color)
+    * There is no provision for dealing with the xcf files generated by GIMP, since darktable doesn't deal with 
+      them.  You may want to save the xcf file if you intend on doing further edits to the image or need to save 
+      the layers used.  Where you save them is up to you.
+
+    BUGS, COMMENTS, SUGGESTIONS
+    * Send to Bill Ferguson, wpferguson@gmail.com
+]]
+
+local dt = require "darktable"
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("gimp",dt.configuration.config_dir.."/lua/")
+
+local function split_filepath(str)
+  local result = {}
+  -- Thank you Tobias Jakobs for the awesome regular expression, which I tweaked a little
+  result["path"], result["filename"], result["basename"], result["filetype"] = string.match(str, "(.-)(([^\\/]-)%.?([^%.\\/]*))$")
+  return result
+end
+
+local function get_path(str)
+  local parts = split_filepath(str)
+  return parts["path"]
+end
+
+local function get_filename(str)
+  local parts = split_filepath(str)
+  return parts["filename"]
+end
+
+local function get_basename(str)
+  local parts = split_filepath(str)
+  return parts["basename"]
+end
+
+local function get_filetype(str)
+  local parts = split_filepath(str)
+  return parts["filetype"]
+end
+
+local function _(msgid)
+    return gettext.dgettext("gimp", msgid)
+end
+
+local function checkIfBinExists(bin)
+  local handle = io.popen("which "..bin)
+  local result = handle:read()
+  local ret
+  handle:close()
+  if (result) then
+    dt.print_error("true checkIfBinExists: "..bin)
+    ret = true
+  else
+    dt.print_error(bin.." not found")
+    ret = false
+  end
+  return ret
+end
+
+local function checkIfFileExists(filepath)
```

Comment:
Just created ~/Pictures/Darktable/test data/path with a space in it/ and populated it with every kind and variation of messed up filename I can think of.  Thanks for the fix.